R/mutate_mzroll_list.R

```diff
@@ -50,12 +50,99 @@ floor_peaks <- function(mzroll_list,
       )),
     missing_peaks
   )
-
+  
   mzroll_list$measurements <- completed_peaks
+  
+  return(mzroll_list)
+}
 
+#' Impute missing peaks with provided feature-level imputation values
+#'
+#'@param mzroll_list: data in triple omic structure
+#'@param lod_values: a tibble that maps groupId to log2 feature-level imputation values
+#'@param quant_var: column to use for peak values
+#'@param imputation_sd: standard deviation of Gaussian distribution to use for missing peak imputation
+#'
+#'@return triple omic data with imputed missing peaks
+#'
+#' @examples
+#' mzroll_list_imputed <- impute_missing_peaks(mzroll_list, lod_values, "log2_abundance", 0.15)
+#'
+#'@export
+impute_missing_peaks <- function(mzroll_list,
+                                 lod_values,
+                                 quant_var = "log2_abundance",
+                                 imputation_sd = 0.15) {
+  
+  test_mzroll_list(mzroll_list)
+  
+  valid_quant_var <- setdiff(
+    mzroll_list$design$measurements$variable,
+    c(mzroll_list$design$feature_pk, mzroll_list$design$sample_pk)
+  )
+  
+  checkmate::assertChoice(quant_var, valid_quant_var)
+  checkmate::assertNumeric(mzroll_list$measurements[[quant_var]])
+  
+  ## find missing peaks
+  missing_peaks <- tidyr::expand_grid(
+    groupId = mzroll_list$features$groupId,
+    sampleId = mzroll_list$samples$sampleId) %>%
+    dplyr::anti_join (
+      mzroll_list$measurements,
+      by = c("groupId", "sampleId")
+    )
+  
+  ## impute missing peaks
+  missing_peaks_imputed <- left_join(
+    missing_peaks, 
+    lod_values, 
```

Comment:
You should perform some quick validation of lod_values. i.e., there is only one groupId per lod value. The list of groupIds is exactly equal to the groupIds in the features table. The columns in the table are as expected (quant_var exists + groupId)

Reply:
@shackett added validation checks on this commit: https://github.com/calico/claman/pull/22/commits/abe5470cb0246a9b65962629f7b1012e346f34f4